bin/maintenance/update_header.py

```diff
@@ -179,31 +181,35 @@ def main():
         sys.exit(1)
 
     project, year, path, file_ = _process_args(sys.argv[1:])
+    error = False
 
     if path is not None:
         print(cformat("Updating headers to the year %{yellow!}{year}%{reset} for all the files in "
                       "%{yellow!}{path}%{reset}...").format(year=year, path=path))
         for root, _, filenames in os.walk(path):
             for filename in filenames:
                 if not blacklisted(path, root):
-                    update_header(project, os.path.join(root, filename), year)
+                    error = update_header(project, os.path.join(root, filename), year) or error
     elif file_ is not None:
         print(cformat("Updating headers to the year %{yellow!}{year}%{reset} for the file "
                       "%{yellow!}{file}%{reset}...").format(year=year, file=file_))
-        update_header(project, file_, year)
+        error = update_header(project, file_, year) or error
     else:
         print(cformat("Updating headers to the year %{yellow!}{year}%{reset} for all "
                       "git-tracked files...").format(year=year))
         try:
             for path in subprocess.check_output(['git', 'ls-files']).splitlines():
                 path = os.path.abspath(path)
                 if not blacklisted(os.getcwd(), os.path.dirname(path)):
-                    update_header(project, path, year)
+                    error = update_header(project, path, year) or error
         except subprocess.CalledProcessError:
             print(cformat('%{red!}[ERROR] you must be within a git repository to run this script.'), file=sys.stderr)
             print(USAGE)
             sys.exit(1)
 
+    if error:
+        sys.exit(1)
```

Comment:
I'd prefer to put this behind a `--ci` flag like we did in `update_backrefs.py`.
Having a non-zero exit code in a script that does its job as intended and without any errors just doesn't seem right.

Reply:
Sounds good 👍